src/main/java/com/exasol/bucketfs/monitor/TimestampState.java

```diff
@@ -0,0 +1,70 @@
+package com.exasol.bucketfs.monitor;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.TimeZone;
+
+import com.exasol.bucketfs.monitor.BucketFsMonitor.State;
+
+/*
+ * COPY from project exasol-testcontainers.
+ * SHOULD be replaced by test dependency.
+ */
+
+/**
+ * Reject other states with earlier time.
+ */
+public class TimestampState implements BucketFsMonitor.State {
+
+    /**
+     * Create a new instance of {@link TimestampState}
+     *
+     * @param time current instant in time before the operation waiting for synchronization.
+     * @return state representing time truncated to low resolution by discarding the micro seconds.
+     */
+    public static TimestampState lowResolution(final Instant time) {
+        return new TimestampState(time.truncatedTo(ChronoUnit.MICROS));
+    }
+
+    /**
+     * Create a new instance of {@link TimestampState}
+     *
+     * @param time     current instant in time represented as {@link LocalDateTime}
+     * @param timeZone time zone
+     * @return time stamp state representing this time
+     */
+    public static TimestampState of(final LocalDateTime time, final TimeZone timeZone) {
+        return new TimestampState(time.atZone(timeZone.toZoneId()).toInstant());
```

Comment:
Do you also need to truncate microseconds?

Reply:
No. 
Background is that precision of timestamps from logs is limited (although I personally assume more than microseconds, I guess at best milliseconds, if we're lucky). In contrast precision of current time is probably higher.
So the initial strategy was 
1. get current time in UTC and truncate resolution
2. get time from logs in local time (inherently with low resolution)
3. only accept log entries with timestamp identical time or later, but not earlier than current time (truncated)